src/modelAnalysis/topology/pathVectors.m

```diff
@@ -0,0 +1,195 @@
+%                   removed in efms if the corresponding reactions
+%                   are not contained in any mode) %
+%
+% rev:              vector indicating for each mode whether it is
+%                   reversible(0)/irreversible (1)
+%
+% idx:              maps the columns in efm onto the column indices
+%                   in cnap.stoichmat, i.e. idx(i) refers to the
+%                   column number in cnap.stoichmat (and to
+%	                  the row number in cnap.reacID)
+%
+% ray:              indicates whether the i-th row (vector) in efm
+%                   is an unbounded (1) or bounded (0) direction
+%                   of the flux cone / flux polyhedron. Bounded
+%                   directions (such as extreme points) can only
+%                   arise if an inhomogeneous problem was defined
+%                   (see also above for 'constraints').
+% cbmodel:          If the input model be changed during computations
+%                   then the new model will be saved as cbmodel (COBRA
+%                   model)
+%
+% .. Author: Susan Ghaderi, LCSB, 06.06.2017
+%
+
+constraints = [];
+positivity = 0;
+mexVersion = 4;
+irrevFlag = 1;
+convBasisFlag = 0;
+isoFlag = 0;
+cMacro = 0;
+efmToolOptions = {};
+display = 'None';
+
+%% varargin checking
+if numel(varargin) > 1
+    for i = 1:2:numel(varargin)
+        key = varargin{i};
+        value = varargin{i + 1};
+        switch key
+            case 'constraints'
+                constraints = value;
+            case 'positivity'
+                positivity = value;
+            case 'mexVersion'
+                mexVersion = value;
+            case 'irrevFlag'
+                irrevFlag = value;
+            case 'convBasisFlag'
+                convBasisFlag = value;
+            case 'isoFlag'
+                isoFlag = value;
+            case 'cMacro'
+                cMacro = value;
+            case 'efmToolOptions'
+                efmToolOptions = value;
+            case 'display'
+                display = value;
+            otherwise
+                msg = sprintf('Unexpected key %s', key)
+                error(msg);
+        end
+
+    end
+end
+
+%% Error checking
+if mexVersion ~= 1 && mexVersion ~= 2 && mexVersion ~= 3 && mexVersion ~= 4
+    error('mexversion should be either 1 or 2 or 3 or 4');
+end
+
+if irrevFlag ~= 0 && irrevFlag ~= 1
+    error('irrev_flag should be either 0 or 1');
+end
+
+if convBasisFlag ~= 0 && convBasisFlag ~= 1
+    error('convbasis_flag should be either 0 or 1');
+end
+
+if isoFlag ~= 0 && isoFlag ~= 1
+    error('iso_flag should be either 0 or 1');
+end
+
+if positivity ~= 0 && positivity ~= 1
+    error('positivity should be either 0 or 1');
+end
+
+%% Convertining reversible reactions into irreversible reactions
+if positivity
+    ind = model.rev == 1;
+    rev = [model.rev; model.rev(ind)];
+    rxns = strcat(model.rxnNames(ind), '-rev');
+    c = [model.c; model.c(ind)];
```

Comment:
CellNetAnalyser assumes that the model objective in a COBRA model is always maximisation. 
We have to either coordinate with the CNA authors, that they consider the osense field, or we have to modify the csense according to the osense field.